pkg/net/host.go

```diff
@@ -129,11 +128,23 @@ func ParseHost(s string) (*Host, error) {
 		isPortSet = true
 	}
 
-	if i := strings.Index(host, "%"); i > -1 {
-		host = host[:i]
+	if host != "" {
+		host, err = normalizeIPv6(host)
```

Comment:
There is no need to be `normalizeIPv6()` as separate function.  Is there any particular reason?

Reply:
normalizeIPv6 here is to strip the enclosing brackets